commands/transaction.go

```diff
@@ -61,52 +93,93 @@ func CheckTransaction(tx *commandspb.Transaction) (*commandspb.InputData, error)
 		errs.AddForProperty("tx.from.pub_key", ErrShouldBeAValidVegaPubkey)
 	}
 
+	// We need the above check to pass, so we verify it's all good.
 	if !errs.Empty() {
 		return nil, errs.ErrorOrNil()
 	}
-	errs.Merge(validateSignature(tx.Signature, tx.GetPubKey()))
-	if !errs.Empty() {
+
+	inputData, inputErrs := checkInputData(tx.Version, tx.InputData, chainID)
+	if !inputErrs.Empty() {
+		errs.Merge(inputErrs)
 		return nil, errs.ErrorOrNil()
 	}
 
-	inputData, errs := checkInputData(tx.InputData)
+	errs.Merge(checkSignature(tx.Signature, tx.GetPubKey(), tx.InputData))
 	if !errs.Empty() {
 		return nil, errs.ErrorOrNil()
 	}
+
 	return inputData, nil
 }
 
-func validateSignature(signature *commandspb.Signature, pubKey string) Errors {
+func checkSignature(signature *commandspb.Signature, pubKey string, rawInputData []byte) Errors {
 	errs := NewErrors()
-	_, err := hex.DecodeString(signature.Value)
+
+	if signature == nil {
+		return errs.FinalAddForProperty("tx.signature", ErrIsRequired)
+	}
+
+	if len(signature.Value) == 0 {
+		errs.AddForProperty("tx.signature.value", ErrIsRequired)
+	}
+	decodedSig, err := hex.DecodeString(signature.Value)
 	if err != nil {
-		return errs.FinalAddForProperty("tx.signature.value", ErrShouldBeHexEncoded)
+		errs.AddForProperty("tx.signature.value", ErrShouldBeHexEncoded)
 	}
 
-	_, err = hex.DecodeString(pubKey)
+	if len(signature.Algo) == 0 {
+		errs.AddForProperty("tx.signature.algo", ErrIsRequired)
+	}
+	algo, err := wcrypto.NewSignatureAlgorithm(signature.Algo, signature.Version)
+	if err != nil {
+		errs.AddForProperty("tx.signature.algo", ErrUnsupportedAlgorithm)
+		errs.AddForProperty("tx.signature.version", ErrUnsupportedAlgorithm)
+	}
+
+	// We need the above check to pass, so we verify it's all good.
+	if !errs.Empty() {
+		return errs
+	}
+
+	decodedPubKey := []byte(pubKey)
+	if IsVegaPubkey(pubKey) {
```

Comment:
What happen if it's not a valid pubkey?

Reply:
I just use the pubkey as it is provided. We don't have support for anything else.

Reply:
But you check if it’s a valid pubkey. If it’s not one it should be handled isn’t it?

Reply:
It's checked earlier here:
https://github.com/vegaprotocol/vega/pull/5866/files#diff-ee1725107e82a8cfa620ce0297f0d97175314976b9b924c23436f44eda7f613cL60